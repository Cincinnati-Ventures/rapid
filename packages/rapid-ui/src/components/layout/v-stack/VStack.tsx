import React from 'react';
import Stack from '../Stack/Stack';
import type { Spacing } from '../../../types';

interface VStackProps
	extends React.DetailedHTMLProps<
		React.HTMLAttributes<HTMLDivElement>,
		HTMLDivElement
	> {
	spacing?: Spacing;
	styles?: string;
}

const VStack = React.forwardRef<HTMLDivElement, VStackProps>(
	({ styles, children, spacing = 'lg', ...rest }, ref) => {
		// This component does not support custom dividers or wrapping (maybe a TODO ?)
		return (
			<Stack
				{...rest}
				ref={ref}
				spacing={spacing}
				direction='column'
				styles={styles}
			>
				{children}
			</Stack>
		);
	},
);

export default VStack;
