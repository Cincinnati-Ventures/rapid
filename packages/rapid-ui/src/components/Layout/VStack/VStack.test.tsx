import React from 'react';
import { render } from '@testing-library/react';
import { expect, it, describe } from 'vitest';
import VStack from './VStack';

describe('VStack', () => {
	it('should render default correctly', () => {
		const { container } = render(<VStack />);
		expect(container).toMatchSnapshot();
	});

	it('should render correctly with non default props', () => {
		const { container } = render(<VStack styles='p-4' spacing='sm' />);
		expect(container).toMatchSnapshot();
	});

	it('should render correctly with children', () => {
		const { container } = render(
			<VStack styles='bg-red-500' spacing='md'>
				<div>
					<p>Hello world!</p>
				</div>
				<div>
					<p>Hello world!</p>
				</div>
				<div>
					<p>Hello world!</p>
				</div>
				<div>
					<p>Hello world!</p>
				</div>
			</VStack>,
		);
		expect(container).toMatchSnapshot();
	});
});
