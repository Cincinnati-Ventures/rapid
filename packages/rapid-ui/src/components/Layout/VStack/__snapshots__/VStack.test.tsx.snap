// Vitest Snapshot v1

exports[`VStack > should render correctly with children 1`] = `
<div>
  <div
    class="flex flex-col space-y-3 bg-red-500 rapid-stack"
  >
    <div>
      <p>
        Hello world!
      </p>
    </div>
    <div>
      <p>
        Hello world!
      </p>
    </div>
    <div>
      <p>
        Hello world!
      </p>
    </div>
    <div>
      <p>
        Hello world!
      </p>
    </div>
  </div>
</div>
`;

exports[`VStack > should render correctly with non default props 1`] = `
<div>
  <div
    class="flex flex-col space-y-1.5 p-4 rapid-stack"
  />
</div>
`;

exports[`VStack > should render default correctly 1`] = `
<div>
  <div
    class="flex flex-col space-y-6 rapid-stack"
  />
</div>
`;
